File: easy/nice_angles.pl
use strict;
use POSIX;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $a = <INFILE> ) {
    next if ( $a =~ m/^\s$/ );
    my $b = floor($a);
    print "$b.";
    $a = ( $a - $b ) * 60;
    $b = floor($a);
    printf( "%02d'", $b );
    $a = ( $a - $b ) * 60;
    $b = floor($a);
    printf( "%02d\"\n", $b );
}

File: easy/nice_angles.rb
File.open(ARGV[0]).each_line do |line|
  a = line.to_f
  b = a.floor
  print b.to_s + '.'
  a = (a - b) * 60
  b = a.floor
  printf("%02d'", b)
  a = (a - b) * 60
  b = a.floor
  printf("%02d\"\n", b)
end
